### PR TITLE
LPS-177830 Add test for comparison operators

### PR DESCRIPTION
**Pull request by LuismiBarcos**
Related ticket [LPS-177830](https://issues.liferay.com/browse/LPS-177830)
___

## Purpose of the PR
Add a test to cover filtering by comparison operators the object entries related to multiple levels of relationships.

**IT GOES ON TOP OF THIS https://github.com/liferay-headless/liferay-portal/pull/1116**

### CODE REVIEW

**Comment by LuismiBarcos** [comment]
ci:test:relevant

**Comment by LuismiBarcos** [comment]
ci:stop

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java [hunk 1]
```diff
@@ -880,6 +1265,28 @@ private ObjectRelationship _addObjectRelationshipAndRelateObjectsEntries(
 		return objectRelationship;
 	}
 
+	private <T> void _assertFilter(
+			String expectedObjectFieldName, T expectedObjectFieldValue,
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java:2258]: Maybe better to use `Serializable expectedObjectFieldValue` instead to keep the same format as in the rest of the test?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java [hunk 2]
```diff
@@ -930,6 +1337,10 @@ private JSONArray _createObjectEntriesJSONArray(
 		return jsonArray;
 	}
 
+	private String _escape(String string) {
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java:2329]: What do you think about creating a unique method `_getFilter`, something like:

```
private String _getFilter(String filterExpr, Object... elements) {
	return URLCodec.encodeURL(String.format(filterExpr, elements));
}
```

and the usage:

```
_assertFilter(
	_OBJECT_FIELD_NAME_1, _OBJECT_FIELD_VALUE_1,
	_getFilter("%s/%s eq '%s'", _objectRelationship1.getName(),
			_OBJECT_FIELD_NAME_2, _OBJECT_FIELD_VALUE_2),
	_objectDefinition1);
```

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:relevant

**Review by ccorreagg — APPROVED** [review]

**Comment by LuismiBarcos** [comment]
Will be sent here https://github.com/liferay-headless/liferay-portal/pull/1120

**Pull request closed without merging**